DATASETS_v2.md

```diff
@@ -0,0 +1,244 @@
+
+Each legend has a unique filename based on its hash. Legends are not stored using JSON, but an binary encoding called [msgpack](https://msgpack.org/index.html) that has equivalent capabilities.
+
+#### `meta/crs/{identifier}.wkt`
+A dataset may contain well-known-text CRS definitions of any of the CRS that are needed to interpret its geometry. The identifier that is part of the filename here should be the same as the `geometryCRS` identifier in the schema.
+
+#### `.sno-table/meta/metadata/dataset.json`
+A dataset may contain other structured metadata files that apply to the entire dataset, for instance XML metadata according to the http://www.isotc211.org/2005/gmd schema. These are stored in the `metadata/dataset.json` file, where they are keyed first by the metadata standard that is being adhered to, and then by the mimetype of the file. 
+
+Here is a truncated example of a `dataset.json` file with only one entry:
+```json
+{
+  "http://www.isotc211.org/2005/gmd": {
+    "text/xml": "<gmd:MD_Metadata …"
+  }
+}
+```
+
+### Features
+
+Every database table row is stored in its own file. It is stored as an array of values plus the name of the legend that should be used to read it. This array is serialised using messagepack (a binary format equivalent to JSON), but for the sake of readability, the example below is shown in JSON.
+
+For instance, a single feature might be stored as the following:
+```json
+[
+  "204b9886d5dbd9fe3a7edb9a7a7dba699b5202f7",  # legend
+  [
+    1445288,
+    "GP0001e61000000101cce1b0dce@7fx8f4Dc0",
+    1260047, 
+    "Pukerua Bay Police Station",
+    "2018-11-05"
+  ]
+]
+```
+
+Decoding a feature works as follows - first, look up the legend with the given name from the `meta/legend/` directory. This will contain a list of column IDs. There will be the same number of column IDs as values, so they can be zipped together into key-value pairs:
+
+```json
+{
+  "500f5ecc-c02c-e8db-052a-84efb4d08a00": 1445288,
+  "b8ae8ff3-1691-7f6f-bbe0-c03c648b6d67": 
+    "GP0001e61000000101cce1b0dce@7fx8f4Dc0",
+  "4324e825-ffed-8aff-ceac-53a7dde7e2f7": 1260047, 
+  "878f8e4e-433a-b7bb-74d5-b360ccfb3607": "Pukerua Bay Police Station",
+  "c8e75111-0506-a898-4d0e-ed1aa8c81280": "2018-11-05"
+}
+```
+
+Finally, the current schema is consulted to find out what the current position and name of the columns with those IDs, so that a database row can be constructed. There is the possibility that some of those columns are no longer part of the schema, in which case these values will be dropped from the feature. There is also the possibility that new columns have been added to the schema since this feature was written, in which cases extra `NULL` values will be inserted into the feature in the appropriate position. The end result will be a feature that conforms to the current database schema - something like the following:
```

Comment:
```suggestion
Finally, the current schema is consulted to find out the current position and name of the columns with those IDs, so that a database row can be constructed. There is the possibility that some of those columns are no longer part of the schema, in which case these values will be dropped from the feature. There is also the possibility that new columns have been added to the schema since this feature was written, in which case extra `NULL` values will be inserted into the feature in the appropriate position. The end result will be a feature that conforms to the current database schema - something like the following:
```

Reply:
Done